cli/src/diff_util.rs

```diff
@@ -804,11 +808,23 @@ pub fn show_diff_stat(
     formatter.with_label("diff", |formatter| {
         let mut total_added = 0;
         let mut total_removed = 0;
-        for stat in &stats {
+        let total_files = stats.len();
+        for mut stat in stats {
             total_added += stat.added;
             total_removed += stat.removed;
             let bar_added = (stat.added as f64 * factor).ceil() as usize;
             let bar_removed = (stat.removed as f64 * factor).ceil() as usize;
+            // replace start of path with ellipsis if the path is too long
+            if stat.path.chars().count() > max_path_length {
+                let start_index = stat
+                    .path
+                    .char_indices()
+                    .rev()
+                    .nth(max_path_length - 2)
```

Comment:
shouldn't it be `max_path_length - 4`? (or maybe we can add `char.len_utf8()` to skip the character at start_index)

Reply:
Yeah, you're right. I found this surprisingly hard. I've fixed that and added tests as you suggested.